frontend/public/components/utils/dropdown.jsx

```diff
@@ -48,15 +48,17 @@ export class DropdownMixin extends React.PureComponent {
     e.preventDefault();
     e.stopPropagation();
 
-    const {onChange, noSelection, title} = this.props;
+    const { items, actionItem, onChange, noSelection, title } = this.props;
 
     if (onChange) {
       onChange(selectedKey, e);
     }
 
+    const newTitle = actionItem && selectedKey === actionItem.actionKey ? actionItem.actionTitle : items[selectedKey];
+
     this.setState({
       selectedKey,
-      title: noSelection ? title : this.props.items[selectedKey],
+      title: noSelection ? title : newTitle,
```

Comment:
@divyanshiGupta 
Actually we might be better off not setting the state at all when an action item is clicked.
Leave it to the parent to control whether or not the clicking of the action deselect the item.
```
if (selectedKey !== actionItem.actionKey) {
  this.setState(...);
}
```

@karthikjeeyar has a use case where the action is to create a new project. But it should never select the action item. It should keep the current selection.

Reply:
Yeah, but I think this is a bit different than Karthik's use case. Even if we let parent set the selectedKey we still need to make sure that the title is set correctly in the dropdown if the parent does set the state for action item. In that case we just need to move this logic to where title is being initialized based on selectedKey.

Reply:
I'm thinking though why would the dropdown ever want to set the title to be the action. Doesn't make sense really. The action is not selectable, it is however invokable. The fact that the application creation flow shows the title to be the same as the action is actually arbitrary. It could very well assign a different title to the dropdown. 

Let's look at the new create secret action coming up. If there is already a secret selected in the dropdown, then the user clicks the `Create New Secret` action, but cancels the dialog. The user would probably expect to see their current selection set and not the title `Create New Secret`. They didn't actually change their selection. They simply invoked the action.

Does the dropdown let us override the title through props? If the user invokes the create application action, we should assign no an empty selection and set the title to whatever message we want to display.

Reply:
Yes, we can override the title by sending a prop to the dropdown. But what do you mean by assign an empty selection?

Reply:
Ya that's what i was thinking.
When in the create app mode, there is no selected application.

Reply:
Even if we don't set `selectedKey` in the dropdown if its an `actionItem` `onChange` call will send that key back and parent sets that same key as `selectedKey` which is again sent as prop to `Dropdown` for making it controlled. This will break the logic of performing an action based on the selected key. We need to think about various use cases and maybe approach this from a fresh perspective. Maybe we need to change the action item APIs to something totally different to handle all the cases.

Reply:
@karthikjeeyar the parent has control of whether to set the selected key to the action item. If they don't want to do this, they can ignore the selection change events for action items.

To your point about the need to perform an action on the selected item. This is why we shouldn't be setting the selection to the action item.

Reply:
@christianvogt This fix handles my use case wherein I don't want to add action Item label to be selected in the project selector.

Reply:
@christianvogt Code looks good to me. If it works for all the use cases then i am fine with the changes.